src/common/telemetry/exception-telemetry-sanitizer.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { escapeRegExp } from 'lodash';
+import { UnhandledErrorTelemetryData } from '../extension-telemetry-events';
+
+export class ExceptionTelemetrySanitizer {
+    private readonly MAX_MESSAGE_CHARS = 300;
+    private readonly MAX_STACK_CHARS = 5000;
+    private readonly EXCLUDED_PROPERTIES = [
+        'http',
+        'html',
+        'target',
+        'url',
+        'path',
+        'snippet',
+        'selector',
+        'elementSelector',
+        'cssSelector',
+    ];
+
+    constructor(private readonly extensionId: string) {}
+
+    public sanitizeTelemetryData(
+        telemetryData: UnhandledErrorTelemetryData,
+    ): UnhandledErrorTelemetryData | undefined {
+        if (telemetryData.message && telemetryData.message.length > this.MAX_MESSAGE_CHARS) {
+            telemetryData.message = telemetryData.message.substring(0, this.MAX_MESSAGE_CHARS);
+        }
+        if (telemetryData.stackTrace && telemetryData.stackTrace.length > this.MAX_STACK_CHARS) {
+            telemetryData.stackTrace = telemetryData.stackTrace.substring(0, this.MAX_STACK_CHARS);
+        }
+
+        const exclusionRegex = this.generateExclusionRegex();
```

Comment:
Generating the regex is both stateless and expensive; it would be good to do it just once, rather than with every event. I'd probably do that by calling this from the constructor and storing the result in a private readonly field.